ntp-udp/src/raw_socket/set_timestamping_options.rs

```diff
@@ -35,13 +35,18 @@ pub(crate) fn set_timestamping_options(
             | SOF_TIMESTAMPING_OPT_ID;
     }
 
+    // Safety:
+    // we have a reference to the socket, so fd is a valid file descriptor for the duration of the call
+    // SOL_SOCKET + SO_TIMESTAMPING expect a *u32 as value, which &options is. Furthermore, we own options
```

Comment:
I did not find documentation for `SO_TIMESTAMPING` easily on DuckDuckGo. Maybe provide a link here with what's required? It's surprising to me that this would be defined as `u32` rather than something like `c_uint`.

Reply:
added documentation link.